geanyctags/src/geanyctags.c

```diff
@@ -205,6 +205,26 @@ static gchar *generate_find_string(GeanyProject *prj)
 }
 
 
+static const gchar *get_base_path(void)
+{
+	static gchar *ret = NULL;
```

Comment:
I'm not particularly fond of the non-reentrant semantic, but it indeed has the advantage of making other code changes simple.

Reply:
Things start making much more sense once you notice the magic word "static", don't they ;-). And yes, it probably won't win any competition in code beautifulness but despite that it makes other code simpler and easier to maintain.

Reply:
@codebrainz Thanks. While I wouldn't care much for a standalone application (such code is used in Geany's tag manager for instance), your argument about dangers of `static` in dynamically loaded libraries is something worth considering so I have rewritten the patch to avoid this problem.